packages/core-components/src/layout/Sidebar/Bar.tsx

```diff
@@ -21,60 +21,72 @@ import classnames from 'classnames';
 import React, { useState, useContext, useRef } from 'react';
 import Button from '@material-ui/core/Button';
 
-import { sidebarConfig, SidebarContext } from './config';
+import {
+  makeSidebarConfig,
+  makeSidebarSubmenuConfig,
+  SidebarConfig,
+  SidebarContext,
+  SidebarConfigContext,
+  SubmenuConfig,
+  SidebarOptions,
+  SubmenuOptions,
+} from './config';
 import { BackstageTheme } from '@backstage/theme';
 import { SidebarPinStateContext, useContent } from './Page';
 import { MobileSidebar } from './MobileSidebar';
 
 /** @public */
 export type SidebarClassKey = 'drawer' | 'drawerOpen';
 
-const useStyles = makeStyles<BackstageTheme>(
-  theme => ({
-    drawer: {
-      display: 'flex',
-      flexFlow: 'column nowrap',
-      alignItems: 'flex-start',
-      position: 'fixed',
-      left: 0,
-      top: 0,
-      bottom: 0,
-      zIndex: theme.zIndex.appBar,
-      background: theme.palette.navigation.background,
-      overflowX: 'hidden',
-      msOverflowStyle: 'none',
-      scrollbarWidth: 'none',
-      width: sidebarConfig.drawerWidthClosed,
-      transition: theme.transitions.create('width', {
-        easing: theme.transitions.easing.sharp,
-        duration: theme.transitions.duration.shortest,
-      }),
-      '& > *': {
-        flexShrink: 0,
-      },
-      '&::-webkit-scrollbar': {
-        display: 'none',
-      },
+const useStyles = ({ sidebarConfig }: { sidebarConfig: SidebarConfig }) =>
+  makeStyles<BackstageTheme>(
```

Comment:
You'd want to pass props to `useStyles` rather than calling `makeStyles` on every render, see https://v4.mui.com/styles/basics/#adapting-based-on-props

Reply:
:+1: